### PR TITLE
Disable autostart by default for PG apps

### PR DESCRIPTION
**Pull request by billyb2**
Also gives the option to do so. More discussion here: 
https://flyio.discourse.team/t/adding-autostart-option-to-pg/2162

TODO: Add a test to preflight

### CODE REVIEW

File: test/preflight/apps_v2_integration_test.go [hunk 1]
```diff
@@ -303,6 +303,51 @@ func TestAppsV2ConfigSave_PostgresSingleNode(t *testing.T) {
...
+
+	result := f.Fly("m list --json -a %s", appName)
+	err = json.Unmarshal(result.StdOut().Bytes(), &machList)
+	if err != nil {
+		f.Fatalf("failed to parse json: %v [output]: %s\n", err, result.StdOut().String())
+	}
+	require.Equal(t, 1, len(machList), "expected exactly 1 machine after launch")
+	firstMachine := machList[0]
+
+	config := firstMachine["config"].(map[string]interface{})
```
Comment (alichay) [inline test/preflight/apps_v2_integration_test.go:325]: this isn't even a nit, just a suggestion to make it easier to read :)

There's a newish API `f.MachinesList(appname)` that returns `[]api.Machine`, and that way you don't have to inline json decoding or type casting

File: test/preflight/apps_v2_integration_test.go [hunk 2]
```diff
@@ -303,6 +303,51 @@ func TestAppsV2ConfigSave_PostgresSingleNode(t *testing.T) {
...
+	if autostart_disabled, ok := config["disable_machine_autostart"]; ok {
+		require.Equal(t, true, autostart_disabled.(bool), "autostart was enabled")
```
Comment (gwuah) [inline test/preflight/apps_v2_integration_test.go:327]: shouldn't this be false? you didn't pass `--autostart` on line 313, so I expect the value to be false.

**Review by gwuah — COMMENTED** [review]
either I don't understand the feature or you swapped the assertions 🤔

File: test/preflight/apps_v2_integration_test.go (hunk 1 above)
Comment (billyb2) [reply test/preflight/apps_v2_integration_test.go:325]: Thank God, I'll change it

File: test/preflight/apps_v2_integration_test.go [hunk 3]
```diff
@@ -303,6 +303,51 @@ func TestAppsV2ConfigSave_PostgresSingleNode(t *testing.T) {
...
+	if err != nil {
+		f.Fatalf("failed to parse json: %v [output]: %s\n", err, result.StdOut().String())
+	}
+	require.Equal(t, 1, len(machList), "expected exactly 1 machine after launch")
+	firstMachine = machList[0]
+
+	config = firstMachine["config"].(map[string]interface{})
+
+	if autostart_disabled, ok := config["disable_machine_autostart"]; ok {
+		require.Equal(t, false, autostart_disabled.(bool), "autostart was enabled")
```
[the remark this answers is not in the document]
Comment (alichay) [reply test/preflight/apps_v2_integration_test.go:347]: this is `disable_machine_autostart` there's a double-negative that makes it confusing to understand, but this seems right.

`--autostart` is passed, which means it should autostart, which means autostart should _not_ be disabled, which means `disable_machine_autostart` should be false.

[the remark this answers is not in the document]
Comment (gwuah) [reply test/preflight/apps_v2_integration_test.go:347]: ah you are both right, reverse boolean logic kicking my ass once again.

**Review by gwuah — APPROVED** [review]
LGTM ✚

File: test/preflight/apps_v2_integration_test.go (hunk 1 above)
Comment (billyb2) [reply test/preflight/apps_v2_integration_test.go:325]: This should be changed in the other tests too, but that's definetely out of scope for this PR

File: test/preflight/apps_v2_integration_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (billyb2) [reply test/preflight/apps_v2_integration_test.go:347]: It's super confusing lol. I'll change the error to be easier to read

File: test/preflight/apps_v2_integration_test.go [hunk 4]
```diff
@@ -61,46 +61,24 @@ func TestAppsV2Example(t *testing.T) {
 		f.Fatalf("GET %s never returned 200 OK response after %d tries; last status code was: %d", appUrl, attempts, lastStatusCode)
 	}
 
-	result = f.Fly("m list --json")
-	var machList []map[string]any
-	err = json.Unmarshal(result.StdOut().Bytes(), &machList)
-	if err != nil {
-		f.Fatalf("failed to parse json: %v [output]: %s\n", err, result.StdOut().String())
-	}
-	require.Equal(t, 1, len(machList), "expected exactly 1 machine after launch")
```
[the remark this answers is not in the document]
Comment (billyb2) [reply test/preflight/apps_v2_integration_test.go:70]: Don't be! That's totally my fault

**Review by alichay — APPROVED** [review]
sorry about the back-and-forth!

**Comment by billyb2** [comment]
Don't be, it's great having fresh sets of eyes on PRs :)

**Pull request merged**